Alg.java

```diff
@@ -0,0 +1,605 @@
+import java.util.PriorityQueue;
+import java.lang.Integer;
+import java.util.ArrayList;
+
+public class Alg {
+   /*Compares how similar the key is (MAX ADDITION: 18) */
+   private int compKey(String key1, String key2) {
```

Comment:
Ok - I see what you're doing here. I fear SonarCube is going to flag this as a code smell because of the large size of the method. 

If it does, a good solution to this problem would be to create a static array with the order of the keys (as string). 
Look up the two keys and given them a score based on the differences in their index into the array.

Ask me if that doesn't make sense